docs/tooling-consolidation.md

```diff
@@ -0,0 +1,104 @@
+
+### Developer efficiency (PR builds)
+
+A concern that has been voiced is that working in aspnetcore would be significantly slower than in aspnetcore-tooling given that the PR validation in aspnetcore takes longer. For comparison, it takes about 30 mins to run builds and tests in aspnetcore-tooling whereas builds and tests in aspnetcore takes about 1.5 - 2 hours. We are making efforts in improving this experience but it will be out of scope of aspnetcore-tooling consolidation.
+
+ Note that this will only apply to the migrated language projects in phase one. In phase two, we will explore several approaches to improve this area. We are considering adopting additiona logic (from dotnet/runtime) that will allow us to run only portions of tests based on what source changes were detected. However, we will always build the entire repository regardless of what source changes were made.
+
+### Testing
+
+In phase one, the migrated projects are C# only so there are no test infrasture changes needed in aspnetcore. The existing vscode jest, and node tests will remain in aspnetcore-tooling during phase one and will be migrated in phase two.
+
+### Reliability
+
+There is concern that there is test reliability issues in both repos and combining them would aggravate the problem in aspnetcore. This has been deemed a non-issue in phase one since the test to be migrated are found to be very reliable. For phase two, we will ensure tests are reliable before we migrate each tooling component over to aspnetcore.
+
+### Release cycles
+
+Tooling release cycles is a cross product of .NET Core (2.1, 3.1, 5.0) and VS (vs-mac, vscode, vs) release cycles. In phase one, the packages we are migrating are only released on .NET Core cycles so it should not add any complexity to servicing in aspnetcore. We will follow the existing branching and servicing strategies. In phase two, we will evaluate branching and release policies for tooling components.
+
+### Cross-repo changes
+
+Given that some of the work in tooling scenarios may involve projects in both aspnetcore and aspnetcore-tooling, there is added complexity here to coordinate cross-repo changes. However, this concern is already present due to the dependency of aspnetcore on aspnetcore-tooling. This problem is not new but will affect different projects after the migration; some current cross-repo changes will be easier while some changes between language and tooling packages will be more burdensome.
+
+## Timeline and costs
+
+Phase one can begin as soon as possible and will likely take about 1 week. Phase two's schedule and costs are TBD.
```

Comment:
Does the tooling repo effectively stop work for that period? If that's the case, would it help to do a dry-run and work all the issues before doing a stop-the-world merge?

Reply:
The overall work for stabilization takes about a week, but it doesn't actually block any ongoing work. After we complete the migration of the sources from aspnetcore-tooling to aspnetcore, we can do a cleanup merge for any changes that were committed in the mean time but will likely ask authors of open PRs to migrate their PRs to aspnetcore. Overall disruption to developer work should be small.